File: src/client/src/app/models/order.ts
import { City } from './city';
import { District } from './district';
import { OrderItem } from './order-item';
import { User } from './user';

export interface Order {
  id: number;
  orderNo: number;
  addressTitle: string;
  address: string;
  zipCode: string;
  total: number;
  city: City;
  district: District;
  appUserId: number;
  user: string;
  orderItems: OrderItem[];
  createdAt: Date;
  appUser: User;
}

File: src/server/Entity/Concrete/Order.cs
using Entity.Concrete.Common;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Entity.Concrete
{
    public class Order : BaseEntity
    {
        public int OrderNo { get; set; }
        public string Address { get; set; }
        public string AddressTitle { get; set; }
        public string ZipCode { get; set; }
        public double Total { get; set; }

        public int CityId { get; set; }
        public City City { get; set; }

        public int DistrictId { get; set; }
        public District District { get; set; }

        public int AppUserId { get; set; }
        public AppUser AppUser { get; set; }

        public ICollection<OrderItem> OrderItems { get; set; }
    }
}
